### DISCUSSION TITLE
API: Retrieve the trace by metdata

### DISCUSSION THREAD

**Discussion by gingerjx** [langfuse/langfuse#3276] [Ideas]
### Describe the feature or potential improvement

Currently, API allows to retrieve by multiple parameters like `id` or `tags`. However, it would be great to fetch the trace using a specific key's value from the trace metadata.

**Comment by baggiponte** [comment]
Huge +1 on this one on my side 😊 I'm currently slamming a lot of stuff in the `tags` section but that feels inappropriate, and I cannot use keys to filter.

As an aside, you cal already filter by metadata in the langfuse UI.

**Reply by marcklingen** [reply]
thanks, looking into this

**Comment by shanewho** [comment]
I just ran into this. The use case is I have a slack bot. I store the slack message ID in langfuse metadata. Now I want user to be able to emoji react thumbs up/down the slack mesage if it's a good/bad response and save that feedback back to the trace. Slack webhooks me the message id, but I don't have any way to look up trace by that id to assign the user feedback (without storing my own mapping in a table). Can query this in the UI but not seeing anything in the API to do it.

**Reply by marcklingen** [reply]
Can you use the "slack message ID" as a traceId? This should solve for this problem.

**Comment by martin-beiro** [comment]
Was there any progress on this?  It would be really useful for complex scoring scenarios where there isnt a 1 to 1 trace_id - thing to score relationship too.  It can be done with tags, but it feels wrong

**Comment by jannikmaierhoefer** [comment]
Hi all, querying traces via API or SDK now allows filering for metadata: https://langfuse.com/changelog/2025-10-10-advanced-filtering-public-traces-api
